File: VisitBosnia/visit_bosnia_mobile/lib/model/transactions/transaction_insert_request.dart
import '../credit_card.dart';

class TransactionInsertRequest {
  int? eventId;
  int? appUserId;
  String? description;
  int? quantity;
  double? price;
  late CreditCard creditCard;

  TransactionInsertRequest({
    this.eventId,
    this.appUserId,
    this.quantity,
    this.price,
    this.description,
    required this.creditCard,
  });

  TransactionInsertRequest.fromJson(Map<String, dynamic> json) {
    eventId = json['eventId'];
    appUserId = json['appUserId'];
    quantity = json['quantity'];
    price = json['price'];
    description = json['description'];
    creditCard = json['creditCard'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['eventId'] = this.eventId;
    data['appUserId'] = this.appUserId;
    data['quantity'] = this.quantity;
    data['price'] = this.price;
    data['description'] = this.description;
    data['creditCard'] = this.creditCard.toJson();
    return data;
  }
}

File: VisitBosnia/VisitBosnia.Model/Requests/TransactionInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VisitBosnia.Model.Requests
{
    public class TransactionInsertRequest
    {

        
        public int EventId { get; set; }
        public int AppUserId { get; set; }
        public string Description { get; set; } = null!;
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public DateTime? Date { get; set; }
        public string? Status { get; set; }
        public CreditCardData creditCard { get; set; } = null!;
    
        public string? ChargeId { get; set; }



    }
}
